In [1]:
import json
import pandas as pd
import torch
from torch.utils.data import DataLoader, Dataset

In [2]:
# Open the dataset
dataset_train = json.load(open("league_ranked_2020_short_for_testing.json"))

In [3]:
# Open the table with which one can convert champion id to champion name
championid_to_name = pd.read_csv("riot_champion.csv")
championid_to_name.tail(3)

Unnamed: 0 version      id  key    name                   title  \
145           0  10.6.1  Zilean   26  Zilean        the Chronokeeper   
146           0  10.6.1     Zoe  142     Zoe  the Aspect of Twilight   
147           0  10.6.1    Zyra  143    Zyra      Rise of the Thorns   

                                                 blurb                 tags  \
145  Once a powerful Icathian mage, Zilean became o...  ['Support', 'Mage']   
146  As the embodiment of mischief, imagination, an...  ['Mage', 'Support']   
147  Born in an ancient, sorcerous catastrophe, Zyr...  ['Mage', 'Support']   

    partype  info.attack  ...  stats.hpregen  stats.hpregenperlevel  \
145    Mana            2  ...            5.5                    0.5   
146    Mana            1  ...            6.5                    0.6   
147    Mana            4  ...            5.5                    0.5   

     stats.mpregen stats.mpregenperlevel stats.crit stats.critperlevel  \
145         11.335                  0.80          0                  0   
146          8.000                  0.65          0                  0   
147         13.000                  0.40          0                  0   

     stats.attackdamage  stats.attackdamageperlevel  \
145              51.640                         3.0   
146              58.000                         3.3   
147              53.376                         3.2   

     stats.attackspeedperlevel  stats.attackspeed  
145                       2.13              0.625  
146                       2.50              0.625  
147                       2.11              0.625  

[3 rows x 40 columns]

# Convert Champion Key into Index Notatation
The Champion key which you can get by `championid_to_name["keys"]` is weird, the Keys are not ascending/descending in regular intervalls.
So instead we convert the Champion key to the index with that The Champion is between 0 and 147.

In [4]:
# Make championid_to_name["key"] into a list
lookuptable = championid_to_name["key"].values.tolist()

In [5]:
# Get example data
data1 = [dataset_train["data"][0][i]["championId"] for i in range(10)]
data1

[7, 350, 266, 517, 110, 235, 81, 127, 84, 421]

In [6]:
# Convert the example data into index notation
data2 = torch.tensor([lookuptable.index(data1[i]) for i in range(10)])
data2

tensor([ 61, 141,   0, 112, 127, 100,  27,  64,   2,  93])

# Convert Champion key into Champion name

In [14]:
championid_to_name.head(3)

Unnamed: 0 version      id  key    name                title  \
0           0  10.6.1  Aatrox  266  Aatrox     the Darkin Blade   
1           0  10.6.1    Ahri  103    Ahri  the Nine-Tailed Fox   
2           0  10.6.1   Akali   84   Akali   the Rogue Assassin   

                                               blurb                  tags  \
0  Once honored defenders of Shurima against the ...   ['Fighter', 'Tank']   
1  Innately connected to the latent power of Rune...  ['Mage', 'Assassin']   
2  Abandoning the Kinkou Order and her title of t...          ['Assassin']   

      partype  info.attack  ...  stats.hpregen  stats.hpregenperlevel  \
0  Blood Well            8  ...            3.0                    1.0   
1        Mana            3  ...            6.5                    0.6   
2      Energy            5  ...            8.0                    0.5   

   stats.mpregen stats.mpregenperlevel stats.crit stats.critperlevel  \
0            0.0                   0.0          0                  0   
1            8.0                   0.8          0                  0   
2           50.0                   0.0          0                  0   

   stats.attackdamage  stats.attackdamageperlevel  stats.attackspeedperlevel  \
0               60.00                         5.0                        2.5   
1               53.04                         3.0                        2.0   
2               62.40                         3.3                        3.2   

   stats.attackspeed  
0              0.651  
1              0.668  
2              0.625  

[3 rows x 40 columns]

In [15]:
championid_to_name.loc[championid_to_name["key"] == data1[8]]["name"]

2    Akali
Name: name, dtype: object

# Convert Data into one hot vector encoding

In [16]:
# Example
torch.eye(150)[data2[8]]

tensor([0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0.])

In [17]:
# convert data
test = torch.eye(148).index_select(dim=0, index=data2)
print(test)
print(test.shape)

tensor([[0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.],
        [1., 0., 0.,  ..., 0., 0., 0.],
        ...,
        [0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 1.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.]])
torch.Size([10, 148])
